data/api/client-server/definitions/client_event_without_room_id.yaml

```
@@ -112,3 +113,23 @@ properties:
             this.
         title: EventContent
         type: object
+      membership:
+        description: |
+          The room membership of the user making the request, at the time of the event.
+
+          This property is the value of the `membership` property of the
+          requesting user's [`m.room.member`](/client-server-api#mroommember)
+          state at the point of the event, including any changes caused by the
+          event. If the user had yet to join the room at the time of the event
+          (i.e, they have no `m.room.member` state), this property is set to
+          `leave`.
+
+          It is recommended that homeservers SHOULD populate this property
```

This feels slightly duplicative because SHOULD effectively already means RECOMMEND.

> [3](https://datatracker.ietf.org/doc/html/rfc2119#section-3). SHOULD   This word, or the adjective "RECOMMENDED", mean that there
>   may exist valid reasons in particular circumstances to ignore a
>   particular item, but the full implications must be understood and
>   carefully weighed before choosing a different course.

Maybe it could be shortened similar to this?

```suggestion
          Homeservers SHOULD populate this property
```

---

good spot, thanks!